src/Firestore/Transaction.php

```diff
@@ -0,0 +1,302 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Firestore;
+
+use Google\Cloud\Core\DebugInfoTrait;
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Firestore\Connection\ConnectionInterface;
+
+/**
+ * Represents a Firestore transaction.
+ *
+ * This class should be accessed inside a transaction callable, obtained via
+ * {@see Google\Cloud\Firestore\FirestoreClient::runTransaction()}.
+ *
+ * Note that method examples, while shown as being called directly for the sake
+ * of brevity, should be called only within the context of a transaction
+ * callable, as noted above.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Firestore\FirestoreClient;
+ * use Google\Cloud\Firestore\Transaction;
+ *
+ * $firestore = new FirestoreClient();
+ * $document = $firestore->document('users/john');
+ * $firestore->runTransaction(function (Transaction $transaction) use ($document) {
+ *     // Manage Transaction.
+ * });
+ * ```
+ */
+class Transaction
+{
+    use SnapshotTrait;
+    use DebugInfoTrait;
+
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var ValueMapper
+     */
+    private $valueMapper;
+
+    /**
+     * @var string
+     */
+    private $transaction;
+
+    /**
+     * @var string
+     */
+    private $database;
+
+    /**
+     * @var WriteBatch
+     */
+    private $writer;
+
+    /**
+     * @param ConnectionInterface $connection A connection to Cloud Firestore.
+     * @param ValueMapper $valueMapper A Firestore Value Mapper.
+     * @param string $database The database name.
+     * @param string $transaction The transaction ID.
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        ValueMapper $valueMapper,
+        $database,
+        $transaction
+    ) {
+        $this->connection = $connection;
+        $this->valueMapper = $valueMapper;
+        $this->database = $database;
+        $this->transaction = $transaction;
+
+        $this->writer = new WriteBatch($connection, $valueMapper, $database, $transaction);
+    }
+
+    /**
+     * Lazily instantiate a Collection with the current Transaction ID applied.
+     *
+     * Collections hold Firestore documents. Collections cannot be created or
+     * deleted directly - they exist only as implicit namespaces. Once no child
+     * documents remain in a collection, it ceases to exist.
+     *
+     * Creating a Collection within a Transaction is required in order to run
+     * queries as part of a Firestore Transaction.
+     *
+     * Example:
+     * ```
+     * $collection = $transaction->collection('users');
+     * ```
+     *
+     * @param string $name The name of the collection.
+     * @return CollectionReference
+     */
+    public function collection($name)
```

Comment:
This is not how any of the other SDKs work and shouldn't be needed since this class has all the required API endpoints.

Please note that one of our goals is to show samples side-by-side, which should result in an overall consistent API experience, no matter what language or environment is being used.  You can look at how similar our other APIs are here: https://firebase.google.com/docs/firestore/query-data/get-data

Reply:
My apologies. I missed a detail in the design document regarding running queries in a transaction.